src/strategies_and_params/replicastrategy.jl

```diff
@@ -142,10 +145,26 @@ end
 
 function replica_stats(rs::AllOverlaps{N,<:Any,<:Any,B}, spectral_states::NTuple{N}) where {N,B}
     # Not using broadcasting because it wasn't inferred properly.
-    # For now implement this assuming only a single spectral state; generalise later
-    vecs = ntuple(i -> only(spectral_states[i]).v, Val(N))
-    wms = ntuple(i -> only(spectral_states[i]).wm, Val(N))
-    return all_overlaps(rs.operators, vecs, wms, Val(B))
+    if num_spectral_states(spectral_states[1]) > 1
+        names = String[]
+        values = Float64[]
+        for j in 1:num_spectral_states(spectral_states[1])
+            vecs = ntuple(i -> spectral_states[i][j].v, Val(N))
+            wms = ntuple(i -> spectral_states[i][j].wm, Val(N))
+            names_j, values_j = all_overlaps(rs.operators, vecs, wms, Val(B))
+            append!(
+                names,
+                [name[1:findlast('_',name)-1]*"_s$j"*
+                name[findlast('_',name):end] for name in names_j]
+            )
```

Comment:
```suggestion
            append!(
                names,
                name[1:findlast('_',name)-1]*"_s$j"*
                name[findlast('_',name):end] for name in names_j
            )
```
I think it should work like this (or maybe you need to put the for-loop expression in regular parentheses), so it doesn't need to allocate a bunch of temporary vectors. Also, indexing into strings is generally not recommended, but I don't immediately see a cleaner solution.

Reply:
`all_overlaps` now generates the full names in the first place.